captum/attr/_core/lime.py

```diff
@@ -0,0 +1,1007 @@
+                            else None
+                        )
+                    if expanded_target is None:
+                        expanded_target = _expand_target(target, len(curr_inputs))
+
+                    model_out = _run_forward(
+                        self.forward_func,
+                        _reduce_list(curr_inputs),
+                        expanded_target,
+                        expanded_additional_args,
+                    )
+                    if (
+                        not expand_inputs
+                        and isinstance(model_out, Tensor)
+                        and model_out.numel() != len(curr_inputs)
+                    ):
+                        assert model_out.numel() == bsz * len(
+                            curr_inputs
+                        ), "Number of outputs is not appropriate, must return"
+                        " one output per example. If forward function returns a"
+                        " scalar per batch, ensure that perturbations_per_eval is"
+                        " set to 1."
+                        expand_inputs = True
+                    outputs.append(
+                        model_out.flatten()
+                        if isinstance(model_out, Tensor)
+                        else torch.tensor([model_out])
+                    )
+
+                    curr_inputs = []
+
+            if len(curr_inputs) > 0:
+                expanded_additional_args = (
+                    _expand_additional_forward_args(
+                        additional_forward_args, len(curr_inputs)
+                    )
+                    if additional_forward_args is not None
+                    else None
+                )
+                expanded_target = _expand_target(target, len(curr_inputs))
+                model_out = _run_forward(
+                    self.forward_func,
+                    _reduce_list(curr_inputs),
+                    expanded_target,
+                    expanded_additional_args,
+                )
+                outputs.append(
+                    model_out.flatten()
+                    if isinstance(model_out, Tensor)
+                    else torch.tensor([model_out])
+                )
+
+                if (
+                    not expand_inputs
+                    and isinstance(model_out, Tensor)
+                    and model_out.numel() != len(curr_inputs)
+                ):
+                    assert model_out.numel() == bsz * len(
+                        curr_inputs
+                    ), "Number of outputs is not appropriate, must return"
+                    " one output per example. If forward function returns a"
+                    " scalar per batch, ensure that perturbations_per_eval is set to 1."
+                    expand_inputs = True
+
+            combined_interp_inps = torch.cat(interpretable_inps)
+            combined_outputs = (
+                torch.cat(outputs)
```

Comment:
does cat here assume batch size == 1?

Reply:
No, it's just expected to be a 1D, which should be the case, since it's flattened above.